autogalaxy/operate/image.py

```diff
@@ -74,24 +76,40 @@ def blurred_image_2d_via_convolver_from(
         ----------
         grid
             The 2D (y,x) coordinates of the (masked) grid, in its original geometric reference frame.
-        convolver
-            The convolver object used perform PSF convolution on 1D numpy arrays.
+        psf
+            The PSF the light object 2D image is convolved with.
         blurring_grid
             The 2D (y,x) coordinates neighboring the (masked) grid whose light is blurred into the image.
         """
 
-        image_2d = self.image_2d_from(grid=grid)
-        blurring_image_2d = self.image_2d_from(grid=blurring_grid)
+        image_2d_not_operated = self.image_2d_from(grid=grid, operated_only=False)
+        blurring_image_2d_not_operated = self.image_2d_from(
+            grid=blurring_grid, operated_only=False
+        )
 
-        return convolver.convolve_image(
-            image=image_2d.binned, blurring_image=blurring_image_2d.binned
+        blurred_image_2d = self._blurred_image_2d_from(
+            image_2d=image_2d_not_operated.binned,
+            blurring_image_2d=blurring_image_2d_not_operated.binned,
+            psf=psf,
+            convolver=convolver,
         )
 
+        try:
+            has_light_profile_operated = self.has_light_profile_operated
+        except NotImplementedError:

```

Comment:
Maybe the default value for has_light_profile_operated should be True rather than raising a NotImplementedError?